volatility3/framework/plugins/windows/mftscan.py

```diff
@@ -0,0 +1,331 @@
+                        {
+                            "creation_time": self.human_date(creation_time_win),
+                            "modified_time": self.human_date(modified_time_win),
+                            "updated_time": self.human_date(altered_time_win),
+                            "accessed_time":  self.human_date(access_time_win),
+                            "allocated_size": self.unpack_data(mft_record, attr_data+40,  'unsigned long long'),
+                            "real_size": self.unpack_data(mft_record, attr_data+48,  'unsigned long long'),
+                            "flags": permissions,
+                            "file_name": file_name,
+                            "name_space": name_space
+                        })
+            
+            # Update Offset for next Attribute
+            attr_offset += attr_len
+
+        return mft_entry
+
+    def _generator(self):
+        rules = yara.compile(sources = signatures)
+
+        layer = self.context.layers[self.config['primary']]
+        for offset, rule_name, name, value in layer.scan(context = self.context, scanner = yarascan.YaraScanner(rules = rules)):
+
+            # For each matching rule try to read 1024 bytes (size of an MFT record) at the offset.
+            try:
+                mft_record = layer.read(offset, 1024, False)
+                mft_entry = self.parse_mft_record(mft_record)
+            except PagedInvalidAddressException:
+                mft_entry = None
+            except Exception as err:
+                vollog.error(err)
+                mft_entry = None
+
+            if mft_entry:
+                vollog.debug(mft_entry)
+
+                # Tree Grid is large and variable
+                si = mft_entry['attributes']['SI']
+                fn = mft_entry['attributes']['FN']
+
+                signature = mft_entry.get('signature', 0)
+                record_number = mft_entry.get('record_number', 0)
+                link_count = mft_entry.get('link_count', 0)
+                permissions = mft_entry.get('flags', '')
+                
+                si_creation_time = si.get('creation_time', '')
+                si_modified_time = si.get('modified_time', '')
+                si_updated_time = si.get('updated_time', '')
+                si_accessed_time = si.get('accessed_time', '')
+
+                yield 0, (
+                    format_hints.Hex(offset),
+                    signature,
+                    record_number,
+                    link_count,
+                    permissions,
+                    'Standard Information',
+                    'N/A',
+                    si_creation_time,
+                    si_modified_time,
+                    si_updated_time,
+                    si_accessed_time)
+
+                for entry in fn:
+                    # As this is variable and may or may not exist
+                    # And could have 0-6 entries lets do it per row.  
+                    yield 0, (
```

Comment:
Since this feels related to the row above, it might be worth using `yield 1` rather than `yield 0`, to make it a subrow.